library/juniper_junos_config.py

```diff
@@ -1054,12 +1054,12 @@ def main():
     junos_module.logger.debug("Step 4 - Determine differences between the "
                               "candidate and committed configuration "
                               "databases.")
-    if diff is True:
+    if diff is True or junos_module._diff:
         diff = junos_module.diff_configuration()
         if diff is not None:
             results['changed'] = True
-            if return_output is True:
-                results['diff'] = diff
+            if return_output is True or junos_module._diff:
+                results['diff'] = {'prepared': diff}
```

Comment:
- Why is a new key `prepared` required here ?? 
- Is it a standard to print generated `diff` using `prepared` as key in Ansible ??

Reply:
As we already have the diff caculated we can use the `prepared`-key to print it out.
See "Ansible Module development : How to add diff support to your module":
https://www.networktocode.com/blog/post/generating-diff-with-ansible/

Reply:
@rsmekala For same blog, I was mentioning about the tweet today morning from @dgarros <blockquote class="twitter-tweet"><p lang="en" dir="ltr">.<a href="https://twitter.com/ansible?ref_src=twsrc%5Etfw">@ansible</a>’s feature --diff has yet to be widely adopted in the network industry. <a href="https://twitter.com/damgarros?ref_src=twsrc%5Etfw">@damgarros</a> explains how to add diff support to your module: <a href="https://t.co/hvoipPFQPI">https://t.co/hvoipPFQPI</a></p>&mdash; Network to Code (@networktocode) <a href="https://twitter.com/networktocode/status/1169241297338753024?ref_src=twsrc%5Etfw">September 4, 2019</a></blockquote>